test/functional/test-amp-img.js

```diff
@@ -374,4 +377,80 @@ describe('amp-img', () => {
     impl.unlayoutCallback();
   });
 
+  describe('blurred image placeholder', () => {
+    let el;
+    let img;
+    let impl;
+
+    beforeEach(() => {
+      getImgWithBlur(true, true);
+      toggleExperiment(impl.win, 'blurry-placeholder', true, true);
+    });
+
+    afterEach(() => {
+      impl.unlayoutCallback();
+    });
+
+    /**
+     * Creates an amp-img with an image child that could potentially be a
+     * blurry placeholder.
+     * @param {boolean} addPlaceholder Whether the child should have a
+     *     placeholder attribute.
+     * @param {boolean} addBlurClass Whether the child should have the
+     *     class that allows it to be a blurred placeholder.
+     */
+    function getImgWithBlur(addPlaceholder, addBlurClass) {
+      el = document.createElement('amp-img');
+      img = document.createElement('img');
+      if (addPlaceholder) {
+        img.setAttribute('placeholder', '');
+      }
+      if (addBlurClass) {
+        img.classList.add('i-amphtml-blur');
+      }
+      el.appendChild(img);
+      el.getResources = () => Services.resourcesForDoc(document);
+      el.getPlaceholder = sandbox.stub();
+      el.togglePlaceholder = sandbox.stub();
+      impl = new AmpImg(el);
+      impl.layoutWidth_ = 200;
+    }
+
+    it('should correctly detect if a blurry image child exists', () => {
+      el.getPlaceholder = () => {return img;};
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      expect(img.style.opacity).to.equal('0');
+
+      getImgWithBlur(true, false);
+      impl.buildCallback();
+      impl.layoutCallback();
+      impl.firstLayoutCompleted();
+      expect(img.style.opacity).to.be.equal('');
```

Comment:
Let's also verify that `togglePlaceholder(false)` continues to be called in the default case, here and below.

Reply:
So is this still a case of testing that the appropriate css styling is given to the element rather than just just asserting that togglePlaceholder was called with false?